src/pickup.cpp

```diff
@@ -214,7 +215,32 @@ bool pick_one_up( item_location &loc, int quantity, bool &got_water, bool &offer
         picked_up = true;
         option = NUM_ANSWERS; //Skip the options part
     } else if( newit.made_of_from_type( LIQUID ) ) {
-        got_water = true;
+
+        //If the item we're trying to pick up is a frozen liquid check if we have a hammering tool and if the user wants to harvest the liquid
+        if( newit.is_frozen_liquid() ) {
+            if( u.has_quality( quality_id( "HAMMER" ) ) ) {
+                item hammering_item = u.item_with_best_of_quality( quality_id( "HAMMER" ) );
+                if( query_yn( _( string_format( "Do you want to crush up %s with your %s?\n" +
+                                                colorize( "Be wary of fragile items nearby!", c_red ), newit.display_name(),
+                                                hammering_item.display_name() ) ) ) ) {
+
+                    //Risk smashing tile with hammering tool, risk is lower with higher dex, damage lower with lower strength
+                    if( one_in( 1 + u.dex_cur / 4 ) ) {
+                        add_msg( _( colorize( ( "You swing your %s wildly!", hammering_item.display_name() ), c_red ) ) );
```

Comment:
Same here. And here the placement of the parameters doesn't even make sense.
```suggestion
                        add_msg( colorize( _( "You swing your %s wildly!" ), c_red ), hammering_item.display_name() );
```